plugins/steve/resource-class.ts

```diff
@@ -155,24 +176,24 @@ export const STATES = {
   waitcheckin:        { color: 'warning', icon: 'tag' },
   waiting:            { color: 'info', icon: 'tag' },
   warning:            { color: 'warning', icon: 'error' },
-};
+} as StateList;
 
-export function getStatesByType(type = 'info') {
+export function getStatesByType() {
   const out = {
     info:    [],
     error:   [],
     success: [],
     warning: [],
-    unknown: [],
-  };
+    unknown: []
+  } as StateInfoForTypes;
 
-  forIn(STATES, (state, stateKey) => {
-    if (state.color) {
-      if (out[state.color]) {
-        out[state.color].push(stateKey);
-      } else {
-        out.unknown.push(stateKey);
-      }
+  forIn(STATES, (state: State, stateKey: string) => {
+    const color = state.color as StateColor;
```

Comment:
There's a few TS mechanisms you can use to tidy up a lot of the 'state' based types.

Given 
```
export enum StateColor {
    info = 'info',
    error = 'error',
    success = 'success',
    warning = 'warning',
    unknown = 'unknown',
}

```
Objects that require properties for each value of the enum 
```
export type StateInfoForTypes = {
    info: string[];
    error: string[],
    success: string[],
    warning: string[],
    unknown: string[]
}

// can be converted to

export type StateInfoForTypes = {
    [key in StateColor]: string[]
}

```
Note - The above highlights an issue where `STATES` contains an object with color `darker`. This could be a geniune bug or we need to also cater for a `darker` state

You can also do things like ...

```
export type States = keyof typeof StateColor;

// So anything that will be a string from that enum can be defined as

export type State = {
    color: States;
    icon: string;
}

// Which makes enums easy peasy with 

const color = StateColor[state.color];

```

Reply:
Hm. The linter seems to be having a problem with

```js
 const color = StateColor[state.color];
```

It gives the error:

```
(property) color?: "info" | "error" | "success" | "warning" | "unknown" | undefined
A const enum member can only be accessed using a string literal.
```

Reply:
I've refactored it like this:

```js
forIn(STATES, (state: StateDetails, stateKey: string) => {
    const color: State = state.color;

    try {
      out[color].push(stateKey);
    } catch(err) {
      out.unknown.push(stateKey);
    }
  });
```